src/rows.c

```diff
@@ -51,22 +51,31 @@ size_t compute_row_size(int actual_num_fields,
 }
 
 
-PyObject *call_converter_function(PyObject *func, char32_t *token)
+PyObject *call_converter_function(PyObject *func, char32_t *token, bool byte_converters)
 {
     Py_ssize_t tokenlen = 0;
     while (token[tokenlen]) {
         ++tokenlen;
     }
+
+    // Convert token to a Python unicode object
     PyObject *s = PyUnicode_FromKindAndData(PyUnicode_4BYTE_KIND, token, tokenlen);
     if (s == NULL) {
-        // fprintf(stderr, "*** PyUnicode_FromKindAndData failed ***\n");
         return s;
     }
+
+    // If byte_converters flag is set, encode prior to calling converter
+    if (byte_converters) {
+        // TODO: Is it safe to use the same PyObject to store the encoded result?
```

Comment:
Yes, so long `s` was borrowed, which it is not :).  You can use `Py_SETREF(s, ...)` which will do the right thing, or use a new variable, and then decref `s`.